networking_nsxv3/plugins/ml2/drivers/nsxv3/agent/nsxv3_agent.py

```diff
@@ -124,99 +192,86 @@ def security_group_delete(self, security_group_id):
     def security_groups_member_updated(self, context, **kwargs):
         o = kwargs["security_groups"]
         o = o if isinstance(o, list) else [o]
-        for id in o:
-            self.security_group_member_updated(id)
+        self.synchronizer.submit_task(0, o,
+                                      self.security_group_member_updated)
 
     # RPC method
     def security_groups_rule_updated(self, context, **kwargs):
         o = kwargs["security_groups"]
         o = o if isinstance(o, list) else [o]
-        for id in o:
-            self.security_group_rule_updated(id)
-
+        self.synchronizer.submit_task(0, o,
+                                      self.security_group_rule_updated)
+
+    def _sync_inventory_full(self):
+        LOG.info("Full synchronization started")
+        self.synchronizer.submit_task(
+            1,
+            self.get_revisions(
+                query=self.rpc.get_security_group_revision_tuples).keys(),
+            self.sync_security_group)
+        self.synchronizer.submit_task(
+            2,
+            self.get_revisions(
+                query=self.rpc.get_qos_policy_revision_tuples).keys(),
+            self.sync_qos)
+        self.synchronizer.submit_task(3, self.get_revisions(
+            query=self.rpc.get_port_revision_tuples).keys(), self.sync_port)
+
+    def _sync_inventory_shallow(self):
+        LOG.info("Shallow synchronization started")
+        sg_query = self.rpc.get_security_group_revision_tuples
+        qos_query = self.rpc.get_qos_policy_revision_tuples
+        port_query = self.rpc.get_port_revision_tuples
+
+        outdated_ips, orphaned_ips = self._sync_get_content(
+            sdk_model=IPSet(), os_query=sg_query)
+        outdated_qos, orphaned_qos = self._sync_get_content(
```

Comment:
Why is this done before the task using it is started? With it having the lowest priority, the information might be out-of-date for a long time.

Reply:
agree, will apply the change